handlers.go

```diff
@@ -239,32 +240,49 @@ func VerifyReset(w http.ResponseWriter, r *http.Request) {
 	fmt.Fprintf(w, "%s", buildResetCompletePage(result.Roll, result.Email))
 }
 
-func GetEmail(w http.ResponseWriter, r *http.Request) {
-	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+func GetHandlerFunc(input, output string) func (w http.ResponseWriter, r *http.Request) {
```

Comment:
Here, if the GET request is for "email", then the input has to be "roll" and vice-versa. Does it not get redundant here? I think only the output argument shall suffice.

Reply:
Yes. The redundancy is intended. 

For eg, we will include more things in the user record (say Name, Contact number) and then we would have a Get call for them. 

Like, `/get/roll/<ROLL>/phone`. Then, we can use this same closure, with an addition to the switch block.

In fact, this shouldn't be a closure. I just realised that the route should be `/get/{input}/{val}`. We can use a single function for all this. Alright, will clean this up.

Reply:
Decorators FTW! This is neat.

Reply:
Oh nice! Awesome.